mountpoint-s3.yaml

```diff
@@ -0,0 +1,47 @@
+package:
+  name: mountpoint-s3
+  version: 1.10.0
+  epoch: 0
+  description: A simple, high-throughput file client for mounting an Amazon S3 bucket as a local file system.
+  copyright:
+    - license: Apache-2.0
+
+environment:
+  contents:
+    packages:
+      - build-base
+      - busybox
+      - ca-certificates-bundle
+      - clang-19
+      - cmake
+      - fuse3
+      - fuse3-dev
+      - libclang-cpp-19
+      - pkgconf
+      - rust
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/awslabs/mountpoint-s3
+      tag: v${{package.version}}
+      expected-commit: 2b0161600400c7ab7ccb1d6811abb9774facef5d
+      recurse-submodules: true
+
+  - runs: |
+      mkdir -p "${{targets.contextdir}}"/usr/bin
+      cargo build --release
+      install -Dm755 target/release/mount-s3 "${{targets.contextdir}}"/usr/bin/mount-s3
+
+  - uses: strip
+
+test:
+  pipeline:
+    - runs: |
+        mount-s3 --version
```

Comment:
Would there be a way to use minio here to create a bucket and mount it? That seems like a solid way to test things

Reply:
I tried but their client is not suitable with any other s3 compatible sdks, I was tried to use it with minio but it failed with an error like:

```bash
[sdk] ❯ mount-s3 --region us-east-2 --endpoint-url http://localhost:9000 my-test-bucket /tmp
Error: Failed to create S3 client

Caused by:
    0: initial ListObjectsV2 failed for bucket my-test-bucket in region us-east-2
    1: Client error
    2: Unknown CRT error
    3: CRT error 1059: aws-c-io: AWS_IO_DNS_INVALID_NAME, Host name was invalid for dns resolution.
Error: Failed to create mount process
```

Reply:
One last try with the IP address of minio instead of the DNS name? e.g. I see this in an unprivileged container:
```
eee5ce3e6889:~# mount-s3 --region=us-east-1 --endpoint-url http://172.17.0.3:9000 myminio /tmp/minio
fuse: device not found, try 'modprobe fuse' first
Error: Failed to create FUSE session
```

Reply:
oh cmon! it worked, wow!

```
[sdk] ❯ mount-s3 --region=us-east-1 --endpoint-url=http://172.17.0.2:9000 myminio /tmp/minio
bucket myminio is mounted at /tmp/minio
[sdk] ❯
```

Reply:
but it didn't work in the tests due to that: https://github.com/s3fs-fuse/s3fs-fuse/issues/647#issuecomment-330398877.

I think it needs privileged container so I'm afraid that we cannot add more tests :/

<img width="1476" alt="Screenshot 2024-10-19 at 16 06 02" src="https://github.com/user-attachments/assets/e95e2e86-e76d-4039-9ac5-84874027a08f">